### PR TITLE
feat: improve performance of lbc eco experiment layout and use FLSS

### PR DESCRIPTION
**Pull request by eddieferrer**
* Migrate `LoanChannelCategoryClimateExperiment.vue` to use FLSS utilities
* Migrate `KivaClassicSingleCategoryCarousel.vue` to use FLSS utilities
* Remove initial query to get eco experiment subcategory id's from urls, we'll just get them from the loan-channel-query-map
* Refactor `LoanChannelCategoryPage.vue` so that it dynamically loads the layout component on prefetch. This prevents having to run through the preFetch of both the control and experiment before deciding which one to show. 
* A little bit of spacing modifications for the eco experiment layout. 
* Change experiment tracking from 'shown' to 'b' to standardize (will update in prod admin)

### CODE REVIEW

File: src/pages/Lend/LoanChannelCategoryClimateExperiment.vue [hunk 1]
```diff
@@ -131,36 +107,51 @@ export default {
...
+		preFetch(config, client, args) {
+			const { route } = args;
+			const { params } = route;
+
+			// Filter routes on route.param.category to get current path
+			const targetedLoanChannelURL = params.category;
+
+			// Isolate targeted loan channel id
+			const targetedLoanChannelID = loanChannelQueryMapMixin.data().loanChannelQueryMap
+				.find(channel => channel.url === targetedLoanChannelURL).id;
```
Comment (dyersituations) [inline src/pages/Lend/LoanChannelCategoryClimateExperiment.vue:119]: Does there need to be optional chaining on `.id`?

File: src/pages/Lend/LoanChannelCategoryClimateExperiment.vue [hunk 2]
```diff
@@ -131,36 +107,51 @@ export default {
...
-				variables: { basketId: this.loanQueryVars.basketId }
-			});
-		} catch (e) {
-			logReadQueryError(e, 'LoanChannelCategoryControl created ecoCategoryPageQuery');
-		}
-
-		// Filter routes on param.category to get current path
 		this.targetedLoanChannelURL = this.$route?.params?.category;
+		this.targetedLoanChannelID = this.loanChannelQueryMap
+			.find(channel => channel.url === this.targetedLoanChannelURL).id;
```
Comment (dyersituations) [inline src/pages/Lend/LoanChannelCategoryClimateExperiment.vue:137]: Does there need to be optional chaining on `.id`?

File: src/pages/Lend/LoanChannelCategoryClimateExperiment.vue (hunk 1 above)
Comment (eddieferrer) [reply src/pages/Lend/LoanChannelCategoryClimateExperiment.vue:119]: I can add these to be extra safe, but going forward these channels should _really_ be in the query-map

File: src/pages/Lend/LoanChannelCategoryPage.vue [hunk 3]
```diff
@@ -114,6 +117,19 @@ export default {
 					),
 					client.query({ query: experimentAssignmentQuery, variables: { id: 'category_loan_bundles' } }),
 				]);
+			}).then(results => {
+				const experimentSettings = results.map(result => result.data.experiment);
+				const ecoLayoutIsShown = experimentSettings
+					.find(setting => setting.id === 'lend_by_category_carousel_layout').version === 'b';
```
Comment (dyersituations) [inline src/pages/Lend/LoanChannelCategoryPage.vue:123]: Optional chaining on `.version`?

**Review by dyersituations — APPROVED** [review]

**Review by mcstover — APPROVED** [review]
Did you have any additions to make to the loan channel query map? Any new flss compatible objects to add? I suspect there won't be any just yet for the eco friendly categories as they all have tags :(

**Comment by eddieferrer** [comment]
> Did you have any additions to make to the loan channel query map? Any new flss compatible objects to add? I suspect there won't be any just yet for the eco friendly categories as they all have tags :(

None yet, they are all tags and or queryStrings

**Pull request merged**